sources/packages/backend/apps/queue-consumers/src/services/student-file/student-file.service.ts

```diff
@@ -29,33 +35,55 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
    * for any viruses.
    * @param uniqueFileName unique filename of the file to perform the virus scan.
    * @param processSummary process summary logs.
-   * @returns boolean true if the file is virus infected, false otherwise.
+   * @returns boolean true if the file is virus infected, false not infected or null otherwise.
    */
   async scanFile(
     uniqueFileName: string,
     processSummary: ProcessSummary,
   ): Promise<boolean> {
     const studentFile = await this.getStudentFile(uniqueFileName);
     if (!studentFile) {
-      throw new NotFoundException(`Student file ${uniqueFileName} not found.`);
+      throw new CustomNamedError(
+        `File ${uniqueFileName} is not found or has already been scanned for viruses. Scanning the file for viruses is aborted.`,
+        FILE_NOT_FOUND,
+      );
     }
+
     const stream = new Readable();
     stream.push(studentFile.fileContent);
     stream.push(null);
-    const isInfected = await this.clamAVService.scanFile(
-      stream,
-      processSummary,
-    );
-    if (isInfected == null) {
-      // If the file could not be scanned for any reason,
-      // update the file scan status to Pending to keep a
-      // track of the files that haven't been scanned yet.
-      await this.updateFileScanStatus(studentFile.uniqueFileName, isInfected);
-      throw new CustomNamedError(
-        `Unable to scan the file ${studentFile.uniqueFileName}`,
-        UNABLE_TO_SCAN_FILE,
+    let isInfected: boolean;
+    let errorMessage = `Unable to scan the file ${uniqueFileName} for viruses.`;
+    let errorName = "";
+    try {
+      isInfected = await this.clamAVService.scanFile(stream);
+      if (isInfected === null) {
+        errorMessage = `${errorMessage} File scanning has failed due to unknown error.`;
+        errorName = FILE_SCANNING_FAILED;
+      }
+    } catch (err) {
+      if (err.code === "ECONNREFUSED") {
+        errorMessage = `${errorMessage} Connection to ClamAV server failed.`;
+        errorName = CONNECTION_FAILED;
+      } else if (err.code === "ENOTFOUND") {
+        errorMessage = `${errorMessage} ClamAV server is unavailable.`;
+        errorName = SERVER_UNAVAILABLE;
+      } else {
+        errorMessage = `${errorMessage} Unknown error.`;
```

Comment:
For unknown errors, we are losing the `error` information that can help with some troubleshooting.
Can we add the below after line 74?
```
processSummary.error(errorMessage, error);
```